tools/toolchains/__init__.py

```diff
@@ -733,12 +733,27 @@ def add_regions(self):
             ))
             self._add_all_regions(regions, "MBED_RAM")
         try:
-            rom_start, rom_size = self.config.rom
-            Region = namedtuple("Region", "name start size")
-            self._add_defines_from_region(
-                Region("MBED_ROM", rom_start, rom_size),
-                suffixes=["_START", "_SIZE"]
-            )
+            # Add all available ROM regions to build profile
+            rom_available_regions = self.config.get_all_rom_regions(True)
+            for key, value in rom_available_regions.items():
+                rom_start, rom_size = value
+                Region = namedtuple("Region", "name start size")
+                self._add_defines_from_region(
+                    Region("MBED_" + key, rom_start, rom_size),
+                    suffixes=["_START", "_SIZE"]
+                )
+        except ConfigException:
+            pass
+        try:
+            # Add all available RAM regions to build profile
+            ram_available_regions = self.config.get_all_ram_regions()
+            for key, value in ram_available_regions.items():
+                ram_start, ram_size = value
+                Region = namedtuple("Region", "name start size")
```

Comment:
I'd recommend removing this if you also go with the commonly declared `Region` I suggested above.